cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java

```diff
@@ -176,6 +168,9 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
             List<String> cmdsList = new ArrayList<>();
             if (xml != null && xml.trim().length() > 0) {
+                if (!xml.endsWith(".xml")) {
+                    throw new CommandFormatException("The --server-config filename must end with .xml.");
+                }
```

Comment:
This is fine as a workaround, but it smells funny that this is necessary. Can you file a JIRA to sort out why it's needed? Just a comment on the existing JIRA explaining why is fine if you know the answer and it's reasonable.

The check should be if (!xml.trim().endsWith(".xml")) or the trimming can just be done once and the result kept. :)